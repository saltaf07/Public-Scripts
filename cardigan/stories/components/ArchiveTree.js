import { storiesOf } from '@storybook/react';
import ArchiveTree from '../../../common/views/components/ArchiveTree/ArchiveTree';
// import Readme from '../../../common/views/components/ArchiveTree/README.md';

const stories = storiesOf('Components', module);

const ArchiveTreeExample = () => {
  return (
    <ArchiveTree
      collection={[
        {
          path: {
            path: 'PPCRI',
            level: 'Collection',
            label: 'PP/CRI',
            type: 'CollectionPath',
          },
          work: {
            id: 'hz43r7re',
            title: 'Francis Crick (1916-2004): archives',
            alternativeTitles: [],
            type: 'Work',
          },
          children: [
            {
              path: {
                path: 'PPCRI/A',
                level: 'Section',
                label: 'PP/CRI/A',
                type: 'CollectionPath',
              },
              work: {
                id: 'we4g7bbz',
                title: 'Personal Material',
                alternativeTitles: [],
                type: 'Work',
              },
              children: [
                {
                  path: {
                    path: 'PPCRI/A',
                    level: 'Section',
                    label: 'PP/CRI/A',
                    type: 'CollectionPath',
                  },
                  work: {
                    id: 'we4g7bbz',
                    title: 'Test Personal Material',
                    alternativeTitles: [],
                    type: 'Work',
                  },
                },
                {
                  path: {
                    path: 'PPCRI/A',
                    level: 'Section',
                    label: 'PP/CRI/A',
                    type: 'CollectionPath',
                  },
                  work: {
                    id: 'we4g7bbz',
                    title: 'Test Personal Material',
                    alternativeTitles: [],
                    type: 'Work',
                  },
                  children: [
                    {
                      path: {
                        path: 'PPCRI/A',
                        level: 'Section',
                        label: 'PP/CRI/A',
                        type: 'CollectionPath',
                      },
                      work: {
                        id: 'we4g7bbz',
                        title: 'Test2 Personal Material',
                        alternativeTitles: [],
                        type: 'Work',
                      },
                      children: [
                        {
                          path: {
                            path: 'PPCRI/A',
                            level: 'Section',
                            label: 'PP/CRI/A',
                            type: 'CollectionPath',
                          },
                          work: {
                            id: 'we4g7bbz',
                            title: 'Test3 Personal Material',
                            alternativeTitles: [],
                            type: 'Work',
                          },
                        },
                      ],
                    },
                    {
                      path: {
                        path: 'PPCRI/A',
                        level: 'Section',
                        label: 'PP/CRI/A',
                        type: 'CollectionPath',
                      },
                      work: {
                        id: 'we4g7bbz',
                        title: 'Test2 Personal Material',
                        alternativeTitles: [],
                        type: 'Work',
                      },
                    },
                    {
                      path: {
                        path: 'PPCRI/A',
                        level: 'Section',
                        label: 'PP/CRI/A',
                        type: 'CollectionPath',
                      },
                      work: {
                        id: 'we4g7bbz',
                        title: 'Test2 Personal Material',
                        alternativeTitles: [],
                        type: 'Work',
                      },
                    },
                  ],
                },
                {
                  path: {
                    path: 'PPCRI/A',
                    level: 'Section',
                    label: 'PP/CRI/A',
                    type: 'CollectionPath',
                  },
                  work: {
                    id: 'we4g7bbz',
                    title: 'Test Personal Material',
                    alternativeTitles: [],
                    type: 'Work',
                  },
                },
              ],
            },
            {
              path: {
                path: 'PPCRI/B',
                level: 'Section',
                label: 'PP/CRI/B',
                type: 'CollectionPath',
              },
              work: {
                id: 'u8wg83kb',
                title: 'Medical Research Council',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/C',
                level: 'Section',
                label: 'PP/CRI/C',
                type: 'CollectionPath',
              },
              work: {
                id: 'zngeejpx',
                title: 'Salk Institute for Biological Studies',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/D',
                level: 'Section',
                label: 'PP/CRI/D',
                type: 'CollectionPath',
              },
              work: {
                id: 'rcv2reyj',
                title: 'Correspondence',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/E',
                level: 'Section',
                label: 'PP/CRI/E',
                type: 'CollectionPath',
              },
              work: {
                id: 'sse7geug',
                title: 'Travels and Meetings',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/F',
                level: 'Section',
                label: 'PP/CRI/F',
                type: 'CollectionPath',
              },
              work: {
                id: 'n4vmzs9j',
                title: 'Doctorate',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/G',
                level: 'Section',
                label: 'PP/CRI/G',
                type: 'CollectionPath',
              },
              work: {
                id: 'edf9m6n4',
                title: 'Notebooks',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/J',
                level: 'Section',
                label: 'PP/CRI/J',
                type: 'CollectionPath',
              },
              work: {
                id: 'u9aq34kg',
                title: 'Correspondence 1976-2004',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/K',
                level: 'Section',
                label: 'PP/CRI/K',
                type: 'CollectionPath',
              },
              work: {
                id: 'kwc2vjhf',
                title: 'Travel and Lectures 1982-2004',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/L',
                level: 'Section',
                label: 'PP/CRI/L',
                type: 'CollectionPath',
              },
              work: {
                id: 'pruqx8fm',
                title: 'Notes, Drafts and Models [1976]-2004',
                alternativeTitles: [],
                type: 'Work',
              },
            },
            {
              path: {
                path: 'PPCRI/M',
                level: 'Section',
                label: 'PP/CRI/M',
                type: 'CollectionPath',
              },
              work: {
                id: 'd9r2ahvy',
                title: 'Publications 1950-2001',
                alternativeTitles: [],
                type: 'Work',
              },
            },
          ],
        },
      ]}
    />
  );
};

stories.add('Archive Tree', ArchiveTreeExample, {
  // readme: { sidebar: Readme },
});
