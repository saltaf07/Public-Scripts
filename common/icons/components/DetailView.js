const DetailView = props => (
  <svg width={24} height={24} {...props}>
    <rect
      width="12"
      height="16"
      x="6"
      y="4"
      fill="none"
      stroke="#E8E8E8"
      strokeWidth="2"
      rx="1"
    />
  </svg>
);

export default DetailView;
