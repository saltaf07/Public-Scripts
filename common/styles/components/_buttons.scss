.btn {
  border-radius: $border-radius-unit;
  text-decoration: none;
  text-align: center;
  transition: all $transition-properties;
  border: 0;
  white-space: nowrap;

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    cursor: pointer;
  }

  &[disabled],
  &.disabled {
    background: color('pewter');
    border-color: color('pewter');
    cursor: not-allowed;
  }

  &.disabled {
    pointer-events: none;
  }

  .icon {
    display: inline-block;
    vertical-align: middle;
    margin-right: $spacing-unit;
  }

  .icon__shape {
    transition: fill $transition-properties;
    fill: currentColor;
  }
}

.btn--primary,
.btn--secondary {
  padding: $spacing-unit 4 * $spacing-unit;
}

.btn--primary {
  background: color('green');
  color: color('white');
  padding: $spacing-unit 4 * $spacing-unit;

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    background: color('black');
  }
}

.btn--secondary,
.btn--tertiary {
  background: color('white');
  color: color('green');

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    background: color('green');
    color: color('white');
  }
}

.btn--secondary {
  border: 2px solid color('green');
}

.btn--tertiary {
  padding: $spacing-unit 2 * $spacing-unit;
  border: 1px solid color('green');
}

.btn--tertiary-black {
  border-color: color('pewter');
  color: color('smoke');
  background: transparent;
  outline: none;

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    border-color: color('white');
    background: color('charcoal');
  }
}
