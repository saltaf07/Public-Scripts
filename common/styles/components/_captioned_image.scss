//* @define captioned-image
.captioned-image {
  margin: 0;
  display: inline-block;
  width: 100%;
  text-align: center;

  .image {
    max-height: 80vh;
    max-width: 100%;

  }

  /* stylelint-disable */
  noscript .image {
    max-height: none;
  }
  /* stylelint-enable */
}

.captioned-image--fit-vh {
  @include respond-to('large') {
    background: color('black');

    .image {
      max-height: 95vh;
      width: auto;
      min-width: auto;
      margin: 0 auto;
    }
  }
}

// Ensure image and caption are left-aligned
.captioned-image--body {
  text-align: left;

  .caption {
    margin-left: 0;
    margin-right: 0;
  }
}

/* stylelint-disable */
.caption {
  text-align: left;
}
/* stylelint-enable */
