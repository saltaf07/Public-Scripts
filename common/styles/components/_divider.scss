//* @define divider

.divider {
  text-align: left;
  border: 0;
}

.divider--black {
  background-color: color('black');
}

.divider--pumice {
  background-color: color('pumice');
}

.divider--charcoal {
  background-color: color('black');
}

.divider--stub {
  width: 60px;
  height: 5px;
}

.divider--keyline {
  height: 1px;
}

.divider--thin {
  height: 2px;
}

.divider--dashed {
  height: 12px;
  background-image: repeating-linear-gradient(to right, color('charcoal'), color('charcoal') 3px, transparent 3px, transparent 6px);
}
