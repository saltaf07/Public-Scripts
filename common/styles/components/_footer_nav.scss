//* @define footer-nav

.footer-nav {
  color: color('white');
}

.footer-nav__brand {
  display: block;
  bottom: 0;
}

.footer-nav__nav {
  border-bottom: 1px solid color('charcoal');
}

.footer-nav__link {
  display: block;
  text-decoration: none;
  transition: color 200ms ease;

  &:hover {
    color: color('green');
  }
}
