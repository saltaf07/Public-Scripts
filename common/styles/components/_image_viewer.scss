// .image-viewer {
//   display: none; // shown with javascript if browser has the required support
// }

// .image-viewer__content {
//   background-color: color('charcoal');
//   position: absolute;
//   display: none;
//   top: 0;
//   left: 0;
//   bottom: 0;
//   right: 0;
//   display: block;
// }

.image-viewer__controls {
  // TODO: keep an eye on https://github.com/openseadragon/openseadragon/issues/1586
  // for a less heavy handed solution to Openseadragon breaking on touch events
  &,
  button,
  a {
    touch-action: none;
  }

  button {
    display: block;
  }

  .icon {
    margin: 0;
  }

  .btn__text {
    @include visually-hidden;
  }

  position: absolute;
  top: 0;
  left: 12px;
  z-index: 1;
  background: color('coal');
  padding: 6px;
}

.image-viewer__image {
  cursor: zoom-in;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

// .image-viewer__launch-button {
//   position: absolute;
//   bottom: 0;
//   right: 50%;
//   transform: translateX(50%) translateY(50%);
//   z-index: 2;

//   .icon {
//     margin: 0;
//   }

//   .btn__text {
//     @include visually-hidden;
//   }
// }

// react-transition-group classes
// .slideup-viewer-btn {
//   opacity: 0;
//   transform: translateX(50%) translateY(250%);
//   transition: transform 700ms;
// }

// .slideup-viewer-btn-entered {
//   opacity: 1;
//   transform: translateX(50%) translateY(50%);
//   transition: transform 700ms, opacity 700ms;
// }
