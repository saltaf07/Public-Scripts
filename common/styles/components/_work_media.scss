.work-media {
  position: relative;
  min-height: 200px;
  height: calc(100vh - 160px);

  @include respond-to(large) {
    height: calc(100vh - 200px);
  }

  .image {
    max-height: 100%;
    max-width: 100%;
    height: auto;
    width: auto;
    align-self: center;
    margin: 0 auto;

    @supports (object-fit: contain) {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }

  .image--noscript {
    display: block;
    margin: auto;
    width: auto;
    height: auto;
  }
}

.work-media__image-container {
  display: flex;
  // stylelint-disable property-no-vendor-prefix
  -ms-flex: 1 0 auto;
  // stylelint-enable property-no-vendor-prefix
  height: 100%;
  max-width: 800px;
  margin: 0 auto;
}
