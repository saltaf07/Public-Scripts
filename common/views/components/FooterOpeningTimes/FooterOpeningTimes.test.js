import FooterOpeningTimes from './FooterOpeningTimes';
import {
  shallowWithTheme,
  mountWithTheme,
} from '../../../test/fixtures/enzyme-helpers';
import { openingTimes } from '../../../test/fixtures/components/opening-times';

describe('FooterOpeningTimes', () => {
  const component = shallowWithTheme(
    <FooterOpeningTimes
      collectionOpeningTimes={openingTimes.collectionOpeningTimes}
    />
  );

  const componentHtml = component.html();
  it('Should match snapshot of FooterOpeningTimes', () => {
    expect(componentHtml).toMatchSnapshot();
  });

  it('Should display opening times name restaurant as Kitchen', () => {
    expect(componentHtml.match('Kitchen '));
  });

  it('Should not display render any opening times if opening times are empty', () => {
    const mockOpeningTimes = {
      collectionOpeningTimes: {
        placesOpeningHours: [
          {
            id: 'WsuaIB8AAH-yNylo',
            order: 5,
            name: 'Shop',
            openingHours: {
              regular: [],
              exceptional: [],
            },
          },
        ],
      },
    };

    const component = mountWithTheme(
      <FooterOpeningTimes
        collectionOpeningTimes={mockOpeningTimes.collectionOpeningTimes}
      />
    );

    const footerOpeningTimes = component.find('ul');
    expect(footerOpeningTimes.length).toEqual(1);
    expect(footerOpeningTimes.find('li').length).toEqual(0);
  });

  it('Should render venue opening times as closed', () => {
    const mockOpeningTimes = {
      collectionOpeningTimes: {
        placesOpeningHours: [
          {
            id: 'WsuaIB8AAH-yNylo',
            order: 5,
            name: 'Shop',
            openingHours: {
              regular: [
                { dayOfWeek: 'Monday', opens: null, closes: null },
                { dayOfWeek: 'Tuesday', opens: null, closes: null },
                { dayOfWeek: 'Wednesday', opens: null, closes: null },
                { dayOfWeek: 'Thursday', opens: null, closes: null },
                { dayOfWeek: 'Friday', opens: null, closes: null },
                { dayOfWeek: 'Saturday', opens: null, closes: null },
                { dayOfWeek: 'Sunday', opens: null, closes: null },
              ],
              exceptional: [],
            },
          },
        ],
      },
    };

    const component = shallowWithTheme(
      <FooterOpeningTimes
        collectionOpeningTimes={mockOpeningTimes.collectionOpeningTimes}
      />
    );

    expect(component.html().match('Shop closed')).toBeTruthy();
  });
});
