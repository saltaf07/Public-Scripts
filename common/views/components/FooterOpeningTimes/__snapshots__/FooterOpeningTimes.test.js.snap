// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterOpeningTimes Should match snapshot of FooterOpeningTimes 1`] = `"<ul class=\\"plain-list no-padding no-margin\\"><li class=\\"Space-sc-4az8w3-0 iJQveI\\">Galleries <time>10:00</time>—<time>18:00</time></li><li class=\\"Space-sc-4az8w3-0 iJQveI\\">Library <time>10:00</time>—<time>20:00</time></li><li class=\\"Space-sc-4az8w3-0 iJQveI\\">Kitchen <time>11:00</time>—<time>21:00</time></li><li class=\\"Space-sc-4az8w3-0 iJQveI\\">Café <time>08:30</time>—<time>21:00</time></li><li class=\\"Space-sc-4az8w3-0 iJQveI\\">Shop <time>09:00</time>—<time>21:00</time></li></ul>"`;
